nginx_installer/nginx_installer.py

```diff
@@ -326,10 +326,16 @@ def _give_log_recs(self, conf_loc, name, log_filename, all_formats, nginx_versio
 
     def _get_nginx_version(self):
         '''calls out to nginx -v to get the nginx version number (eg 1.4.2)'''
-        verstring = subprocess.check_output(["nginx", "-v"], stderr=subprocess.STDOUT)
-        version = verstring.split()[2]
-        vernum = version.split("/")[1]
-        return vernum
+        try:
+            verstring = check_output(["nginx", "-v"], stderr=subprocess.STDOUT)
+            version = verstring.split()[2]
+            vernum = version.split("/")[1]
+            return vernum
+        except subprocess.CalledProcessError as e:
+            self.warn("error checking nginx version (`{}`), exit status {}".format(e.cmd, e.returncode))
+            self.warn("output:")
+            self.warn(e.output)
+            return None
```

Comment:
this will cause the caller to fail with a `TypeError: expected string or buffer`.
(via https://github.com/honeycombio/honey_installers/blob/master/nginx_installer/nginx_installer.py#L300)

can we also clean up `_give_log_recs` to fail gracefully there (probably with some sort of default)?